### PR TITLE
[AIRFLOW-8902] fix Dag Run UI execution date with timezone cannot be saved issue

### PR DESCRIPTION
**Pull request by realradical**
This PR is to address the following bug: https://github.com/apache/airflow/issues/8842
Airflow 1.10.10 displays datetime with timezone on the web UI. Currently, when user tries to create a dag run on the UI, it gives invalid DateTime error which prevents the dag run being created.
image

---
Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Target Github ISSUE in description if exists
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Comment by realradical** [comment]
@ashb sorry, accidentally closed https://github.com/apache/airflow/pull/8875

added the logic to parse datetime without TZ, and more test cases, please check if it is good

File: dev/airflow-jira [hunk 1]
```diff
@@ -20,11 +20,11 @@
...
 # https://github.com/apache/spark/blob/master/dev/merge_spark_pr.py
 
-from collections import defaultdict, Counter
-
-import jira
 import re
 import sys
+from collections import Counter, defaultdict
+
+import jira
```
Comment (ashb) [inline dev/airflow-jira]: Could you revert the change to this file please as it's unrelated

File: airflow/www/forms.py [hunk 2]
```diff
@@ -17,25 +17,66 @@
...
+                else:
+                    self.data = pendulum.parse(date_str)
+            except ValueError:
+                self.data = None
+                raise ValueError(self.gettext('Not a valid datetime value'))
+
+
 class DateTimeForm(FlaskForm):
     # Date filter form needed for task views
     execution_date = DateTimeField(
```
Comment (ashb) [inline airflow/www/forms.py]:
```suggestion
    execution_date = DateTimeWithTimezoneField(
```

And a few other places in this file too, no?

**Review by ashb — CHANGES_REQUESTED** [review]
At the end of airflow/www/utils.py we have

```

# This class is used directly (i.e. we cant tell Fab to use a different
# subclass) so we have no other option than to edit the converstion table in
# place
FieldConverter.conversion_table = (
    (('is_utcdatetime', DateTimeField, AirflowDateTimePickerWidget),) +
    FieldConverter.conversion_table
)
```

That will be the field that SQLA uses for DT fields -- we should update that to use this new class too.

**Comment by realradical** [comment]
> At the end of airflow/www/utils.py we have
> 
> ```
> 
> # This class is used directly (i.e. we cant tell Fab to use a different
> # subclass) so we have no other option than to edit the converstion table in
> # place
> FieldConverter.conversion_table = (
>     (('is_utcdatetime', DateTimeField, AirflowDateTimePickerWidget),) +
>     FieldConverter.conversion_table
> )
> ```
> 
> That will be the field that SQLA uses for DT fields -- we should update that to use this new class too.

good catch!

File: airflow/www/forms.py (hunk 2 above)
Comment (realradical) [reply airflow/www/forms.py]: good catch, do you think it is good now? @ashb

**Review by ashb — APPROVED** [review]
